File: src/main/java/com/fatec/destino/util/model/usuario/perfil/UserRole.java
package com.fatec.destino.util.model.usuario.perfil;

import java.util.List;

public enum UserRole {
    ADMINISTRADOR(List.of(
          UsuarioAuthority.DELETAR_USUARIO
    )),
    FUNCIONARIO(List.of(
            UsuarioAuthority.CRIAR_PACOTE
    )),
    USUARIO(List.of(
            UsuarioAuthority.EDICAO_PERFIL
    ));

    private final List<UsuarioAuthority> authorities;

    UserRole(List<UsuarioAuthority> authorities) {
        this.authorities = authorities;
    }

    public List<UsuarioAuthority> getAuthorities() {
        return authorities;
    }
}

File: src/main/kotlin/com/fatec/destino/util/model/usuario/perfil/UserRole.kt
package com.fatec.destino.util.model.usuario.perfil

import org.springframework.security.core.GrantedAuthority

enum class UserRole : GrantedAuthority {
    ADMINISTRADOR,
    FUNCIONARIO,
    USUARIO;

    override fun getAuthority(): String {
        return "ROLE_$name"
    }
}